torch/csrc/jit/serialization/python_print.cpp

```diff
@@ -1066,26 +1067,33 @@ struct PythonPrintImpl {
       case prim::CallMethod: {
         const auto& self = node->inputs().at(0);
         const auto& methodName = node->s(attr::name);
-        stmt << "(" << useOf(self) << ")"
-             << "." << methodName << "(";
-        for (size_t i = 1; i < node->inputs().size(); i++) {
-          stmt << useOf(node->inputs()[i]) << ", ";
-        }
-        stmt << ")";
-
         if (auto selfClass = self->type()->cast<ClassType>()) {
           deps_table_.add(selfClass);
-          const Function& method = selfClass->getMethod(node->s(attr::name));
-          TORCH_INTERNAL_ASSERT(
-              method.qualname() ==
-              QualifiedName(selfClass->name()->qualifiedName(), methodName));
+          // check if the method is overloaded and if it is, we serialize
+          // only the qualified name
+          Function* method = selfClass->findMethod(methodName);
+          if (auto overloadedMethod =
+                  dynamic_cast<OverloadedFunction*>(method)) {
+            stmt << "(" << useOf(self) << ")"
+                 << "." << method->qualname().name() << "(";
```

Comment:
Is this unique for each overloaded variant?

Reply:
This will go through sugared_value.h again to do schema matching. Therefore, it doesn't matter if the name is not unique i think